tests/Unit/inc/Engine/CDN/CDN/rewrite.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace WP_Rocket\Tests\Unit\inc\Engine\CDN\CDN;
+
+use Brain\Monkey\Functions;
+use WP_Rocket\Engine\CDN\CDN;
+
+/**
+ * @covers \WP_Rocket\Engine\CDN\CDN::rewrite
+ * @group  CDN
+ */
+class Test_Rewrite extends TestCase {
+
+	/**
+	 * @dataProvider providerTestData
+	 */
+	public function testShouldRewriteURLToCDN( $home_url, $original, $expected ) {
+		Functions\when( 'content_url' )->justReturn( "{$home_url}/wp-content/" );
+		Functions\when( 'includes_url' )->justReturn( "{$home_url}/wp-includes/" );
+		Functions\when( 'wp_upload_dir' )->justReturn( "{$home_url}/wp-content/uploads/" );
+		Functions\when( 'home_url' )->justReturn( $home_url );
+		Functions\when( 'rocket_add_url_protocol' )->alias( function( $url ) {
+			if ( strpos( $url, 'http://' ) !== false || strpos( $url, 'https://' ) !== false ) {
+				return $url;
+			}
+
+			if ( substr( $url, 0, 2 ) === '//' ) {
+				return 'http:' . $url;
+			}
+
+			return 'http://' . $url;
+		} );
```

Comment:
Let's create a base test case for the module and move these stubs into there. It'll reduce the code in the individual tests.

Reply:
they all depend on values passed from the fixtures to the test itself, that's why I didn't move them into the testCase.

Reply:
Good point!